src/main/drivers/accgyro/accgyro_imuf9001.c

```diff
@@ -0,0 +1,351 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "platform.h"
+
+#include "sensors/gyro.h"
+#include "accgyro.h"
+#include "accgyro_mpu.h"
+#include "accgyro_imuf9001.h"
+
+#include "common/axis.h"
+#include "common/maths.h"
+#include "drivers/bus_spi.h"
+#include "drivers/dma_spi.h"
+#include "drivers/exti.h"
+#include "drivers/io.h"
+#include "drivers/sensor.h"
+#include "drivers/time.h"
+#include "fc/config.h"
+#include "fc/runtime_config.h"
+
+#include "sensors/boardalignment.h"
+#include "sensors/gyro.h"
+#include "sensors/acceleration.h"
+
+#include "drivers/system.h"
+
+
+volatile uint16_t imufCurrentVersion = IMUF_FIRMWARE_VERSION;
+volatile uint32_t isImufCalibrating = 0;
+volatile imuFrame_t imufQuat;
+
+void crcConfig(void)
+{
+    return;
+}
+
+inline uint32_t getCrcImuf9001(uint32_t* data, uint32_t size)
+{
+    CRC_ResetDR(); //reset data register
+    for(uint32_t x=0; x<size; x++ )
+    {
+        CRC_CalcCRC(data[x]);
+    }
+    return CRC_GetCRC();
+}
+
+inline void appendCrcToData(uint32_t* data, uint32_t size)
+{
+    data[size] = getCrcImuf9001(data, size);;
+}
+
+static inline void gpio_write_pin(GPIO_TypeDef * GPIOx, uint16_t GPIO_Pin, gpioState_t pinState)
```

Comment:
Compiler is good enough to figure out when to inline this on it's own.
`inline` is just a recommendation anyway.